hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -831,7 +1181,7 @@ object LoadVCF {
               val vcfLine = new VCFLine(line, arrayElementsRequired)
               rvb.start(t.physicalType)
               rvb.startStruct()
-              present = vcfLine.parseAddVariant(rvb, rgBc.map(_.value), contigRecoding, skipInvalidLoci)
+              present = vcfLine.parseAddVariant(rvb, rgBc.map(_.value), contigRecoding, hasRSID, skipInvalidLoci)
```

Comment:
this feels like the wrong place for hasRSID - shouldn't it be handled in the same place as the filter/qual?

Reply:
Is the issue that you have to parse the ID before REF and ALT?